docs/containers/docker-compose.md

```diff
@@ -193,6 +193,76 @@ volumes:
+
+Note: If your workspace has `docker-compose.yml` and `docker-compose.override.yml` and no other compose files, then the `docker-compose` command will be invoked with no input files and it will implicitly use these files. In this case, no customization is needed.
+### Command Customization
+[Command customization](https://code.visualstudio.com/docs/containers/reference#_command-customization) provides various ways to customize the compose up based on your requirement. The following are few sample command customization for the compose up command.
+#### Base file and an override file
+Let's assume your workspace has a base compose file (`docker-compose.yml`) and an override file for each environment (`docker-compose.dev.yml`, `docker-compose.test.yml` and `docker-compose.prod.yml`) and you always compose up with the base file and an override file. In this case the compose up command can be customized like below. When the compose up command is invoked the ${configuration} will be replaced by the selected file.
+```json
+"docker.commands.composeUp": [
+    {
+        "label": "override",
+        "template": "docker-compose -f docker-compose.yml ${configurationFile}  up -d --build",
+    }
+]
+```
+#### Template matching
+Let's assume you have  different set of input files for each environment. You could define multiple templates with regular expression match, and the selected file name will be matched against this `match` property and the corresponding template will be used.
+```json
+"docker.commands.composeUp": [
+    {
+        "label": "dev-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.debug.yml -f docker-compose.dev.yml up -d --build",
+        "match": "dev"
+    },
+    {
+        "label": "test-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.debug.yml -f docker-compose.test.yml up -d --build",
+        "match": "test"
+    },
+    {
+        "label": "prod-match",
+        "template": "docker-compose -f docker-compose.yml -f docker-compose.release.yml -f docker-compose.prod.yml up -d --build",
+        "match": "prod"
+    }
+]
+```
+
+#### Pick a template at runtime
+When invoking the compose up command with the following command customization, you will give an option to choose a template at runtime and the `${configurationFile}` will be replaced with the selected file.
```

Comment:
```suggestion
By invoking the `compose up` command with the following command customization, you can give an option to choose a template at runtime and the `${configurationFile}` will be replaced with the selected file.
```

Reply:
If that's right, I'm not sure who is giving the option to choose a template at runtime - the user?